File: finance-control-mobile/src/domain/currency/currency.ts
/**
 * Represents currency used in application
 */
interface Currency {
  /**
   * Name of the currency
   */
  name: string;

  /**
   * Code of the currency. For example, Dollar is USD
   */
  code: string;

  /**
   * Symbol of the currency. For example, Dollar is $
   */
  symbol: string;
}

export default Currency;

File: profitocracy-ios/Profitocracy/Models/Domain/Entities/Currency.swift
//
//  Currency.swift
//  Profitocracy
//
//  Created by Anton Gavrilov on 28.04.23.
//

import Foundation

struct Currency: Identifiable, Hashable, Codable {
    var id: UUID
    var name: String
    var code: String
    var symbol: String
    
    init(id: UUID = UUID(), name: String, code: String, symbol: String) {
        self.id = id
        self.name = name
        self.code = code
        self.symbol = symbol
    }
}

extension Currency {
    static let availableCurrencies: [Currency] = [
        Currency(name: "US Dollar", code: "USD", symbol: "$"),
        Currency(name: "Russian Ruble", code: "RUB", symbol: "₽"),
        Currency(name: "European Euro", code: "EUR", symbol: "€"),
        Currency(name: "Armenian Dram", code: "AMD", symbol: "֏"),
        Currency(name: "Georgian Lari", code: "GEL", symbol: "₾")
    ]
}
